src/sentry/api/endpoints/organization_discover_query.py

```diff
@@ -88,27 +87,28 @@ def __init__(self, *args, **kwargs):
     def validate(self, data):
         data['arrayjoin'] = self.arrayjoin
 
-        return data
-
-    def validate_range(self, attrs, source):
-        has_start = bool(attrs.get('start'))
-        has_end = bool(attrs.get('end'))
-        has_range = bool(attrs.get('range'))
-
-        if has_start != has_end or has_range == has_start:
+        if data.get('start') or data.get('end'):
```

Comment:
I think this block would read a bit more clearly with an `elif` statement instead of nesting ifs, since the first statement must always be true if the second statement is true
e.g.
```
if data.get('start') and data.get('end'):
    raise .....
elif data.get('start') or data.get('end'):
    return data
```